File: April-Challenges-2021/Day-7/Determine-if-String-Halves-Are-Alike.cpp
class Solution {
public:
    bool isvowel(char a){
        
         if(a == 'a' || a == 'e' || a == 'i' || a == 'o' || a == 'u')
             return 1;
        
         return 0;
        
    }
    bool halvesAreAlike(string s) {
        
         int cnt1 = 0, cnt2 = 0,n = s.size();
        
         for(int i = 0; i < n; i++){
             
             if(isvowel( tolower(s[i]) ) && i < n/2) cnt1++;
             else if(isvowel(tolower(s[i]))) cnt2++;
             
         }
        
         return cnt1 == cnt2;
        
    }
};

File: April-Challenges-2021/Day-7/Determine-if-String-Halves-Are-Alike.py
class Solution:
    def halvesAreAlike(self, s: str) -> bool:
        a,b = s[:len(s)//2],s[len(s)//2:]
        ac,bc = 0,0
        vowels = ['a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U']

        for i in a:
            if i in vowels:
                ac+=1

        for i in b:
            if i in vowels:
                bc+=1
                
        return ac==bc  
